src/main/java/cubicchunks/server/chunkio/IONbtReader.java

```diff
@@ -178,6 +178,9 @@ private static void readEntities(NBTTagCompound nbt, ICubicWorldServer world, Cu
 			}
 
 			entity.addedToChunk = true;
+			entity.chunkCoordX = cube.getX();
```

Comment:
A comment for why it's needed would be great, so that noone tries to remove it again because "it will be set when ticking entity"